### PR TITLE
Add CI/CD

### PR DESCRIPTION
**Pull request by tverum**
Fix issue #1 

Adding github actions workflow for 
- Running automated tests
- Linting (ruff)
- Formatting checks (ruff)

If I understand correctly, the workflow is: 
- Trigger: On PR to main
- Install dependencies
- Run tests
- Run linting
- Run formatting
- Tag commit (semver)
- Test deploy
- Deploy

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 [project]
 name = "lactate-thresholds"
-version = "0.1.0"
+version = "v0.0.1"
 description = "Add your description here"
 readme = "README.md"
 authors = [
-    { name = "Bart6114", email = "bartsmeets86@gmail.com" }
+    { name = "Bart6114", email = "bartsmeets86@gmail.com" },
```
Comment (bart6114) [inline pyproject.toml:7]: can you lowercase me?

File: pyproject.toml [hunk 2]
```diff
@@ -31,6 +32,9 @@ dev = [
     "ruff>=0.8.1",
 ]
 
+[tool.uv]
+trusted-publishing = "always"
```
Comment (bart6114) [inline pyproject.toml:36]: i think the default is automatic, doesn't that default to trusted when in ci?

**Review by bart6114 — COMMENTED** [review]
Plz squash the CI testing stuff and then merge to main.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        run: uv sync --all-extras --dev
+
+      - name: Build the project
+        run: uv build
+
+      - name: Publish to TestPyPi
+        run: uv publish --publish-url https://test.pypi.org/legacy/
+
+      - name: Publish to PyPi
+        if: github.event.repository.fork == false && github.ref =='refs/head/main'
```
[the remark this answers is not in the document]
Comment (tverum) [reply .github/workflows/ci.yaml:91]: I would say no.
You could still be on the main branch of a forked repo right?

File: pyproject.toml [hunk 4]
```diff
@@ -1,10 +1,11 @@
 [project]
 name = "lactate-thresholds"
-version = "0.1.0"
+version = "v0.0.1"
```
[the remark this answers is not in the document]
Comment (tverum) [reply pyproject.toml:3]: Given that in this action run https://github.com/bart6114/lactate-thresholds/actions/runs/12387711956/job/34577620256, we first write the version tag back to the pyproject.toml, and then the build process picks up on the correct version, I would say yes? 

I do not find explicit documentation on the topic though (might be due to lack of correct phrasing).

File: pyproject.toml (hunk 1 above)
Comment (tverum) [reply pyproject.toml:7]: Consider it done.

File: pyproject.toml (hunk 2 above)
Comment (tverum) [reply pyproject.toml:36]: Yes, this is a debugging artefact, from trying to figure out why the trusted-publishing didn't work. 
Removing this.

File: .github/workflows/ci.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tverum) [reply .github/workflows/ci.yaml:91]: Although I am not sure if the `github.event.repository.fork == false` check is entirely necessary. 
Shouldn't the behaviour in the forked repository be defined by the individual/organization forking the repository anyways? 

Source of this is https://github.com/datarootsio/cheek/blob/b721a8011c7e77d98dfdfa342e959c272d9a2c45/.github/workflows/ci.yml#L58 btw ;-)

**Pull request merged**